engine/src/AresException.h

```diff
@@ -0,0 +1,34 @@
+//
+// Created by tdancois on 8/5/16.
+//
+
+#ifndef ARES_EXCEPTION_H
+#define ARES_EXCEPTION_H
+
+
+#include <exception>
+#include <string>
+
+class AresException {
+    std::string emiter;
+    std::string method;
+    std::string message;
+public:
+    AresException(const std::string &message, const std::string &emiter = "", const std::string &method = "");
+
+    virtual std::string what() const;
+
+    const std::string &getEmiter() const;
+
+    const std::string &getMethod() const;
+
+    const std::string &getMessage() const;
+};
+
+class EntityManagerException : public AresException {
+public:
+    EntityManagerException(const std::string &message, const std::string &method = "");
+};
+
+
```

Comment:
Having information about where the program fails is useful for debugging, not for handling the error in case of communication problem for example. (My exception class doesn't help neither.)